### PR TITLE
feat(pytest): add support for buddy.works

### PR DESCRIPTION
**Pull request by juan-fernandez**
## Description
Update CI Visibility metadata extraction used in `pytest`: 

* Add support for [buddy.works](https://buddy.works/)

## Checklist
- [x] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [x] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation
Allow datadog to detect tests run in buddy.works.

## Testing strategy
Unit tests

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
lgtm for after #4881 gets merged

**Review by brettlangdon — APPROVED** [review]

**Review by albertvaka — APPROVED** [review]

**Pull request merged**
